server/functions/make_timestamp.go

```diff
@@ -40,41 +42,77 @@ var make_timestamp = framework.Function6{
-		}
-		day := val3.(int32)
-		if day < 1 || day > 31 {
-			return nil, errDateFieldOutOfRange
-		}
-		hour := val4.(int32)
-		if hour < 0 || hour > 23 {
-			return nil, errTimeFieldOutOfRange
-		}
-		minute := val5.(int32)
-		if minute < 0 || minute > 59 {
-			return nil, errTimeFieldOutOfRange
-		}
-		partialSecond := val6.(float64)
-		second := int(partialSecond)
-		if second < 0 || second >= 60 {
-			return nil, errTimeFieldOutOfRange
+		loc, err := GetServerLocation(ctx)
+		if err != nil {
+			return time.Time{}, err
 		}
-		nsec := int64(partialSecond*float64(time.Second)) % int64(time.Second)
+		return getTimestampInServerLocation(val1.(int32), val2.(int32), val3.(int32), val4.(int32), val5.(int32), val6.(float64), loc)
+	},
+}
 
+// make_timestamptz_int32_int32_int32_int32_int32_float64 represents the PostgreSQL function of the same name, taking the same parameters.
+var make_timestamptz_int32_int32_int32_int32_int32_float64 = framework.Function6{
+	Name:               "make_timestamptz",
+	Return:             pgtypes.TimestampTZ,
+	Parameters:         [6]*pgtypes.DoltgresType{pgtypes.Int32, pgtypes.Int32, pgtypes.Int32, pgtypes.Int32, pgtypes.Int32, pgtypes.Float64},
+	IsNonDeterministic: true,
+	Strict:             true,
+	Callable: func(ctx *sql.Context, _ [7]*pgtypes.DoltgresType, val1, val2, val3, val4, val5, val6 any) (any, error) {
 		loc, err := GetServerLocation(ctx)
 		if err != nil {
-			return nil, err
+			return time.Time{}, err
+		}
+		return getTimestampInServerLocation(val1.(int32), val2.(int32), val3.(int32), val4.(int32), val5.(int32), val6.(float64), loc)
+	},
+}
+
+// make_timestamptz_int32_int32_int32_int32_int32_float64_text represents the PostgreSQL function of the same name, taking the same parameters.
+var make_timestamptz_int32_int32_int32_int32_int32_float64_text = framework.Function7{
+	Name:               "make_timestamptz",
+	Return:             pgtypes.TimestampTZ,
+	Parameters:         [7]*pgtypes.DoltgresType{pgtypes.Int32, pgtypes.Int32, pgtypes.Int32, pgtypes.Int32, pgtypes.Int32, pgtypes.Float64, pgtypes.Text},
+	IsNonDeterministic: true,
+	Strict:             true,
+	Callable: func(ctx *sql.Context, _ [8]*pgtypes.DoltgresType, val1, val2, val3, val4, val5, val6, val7 any) (any, error) {
+		tz := val7.(string)
+		loc, _, _, err := convertTzToOffsetSecs(time.Now().UTC(), tz)
+		if err != nil {
+			return nil, errors.Errorf(`time zone "%s" not recognized`, tz)
 		}
-		return time.Date(int(year), time.Month(month), int(day), int(hour), int(minute), second, int(nsec), loc), nil
+		return getTimestampInServerLocation(val1.(int32), val2.(int32), val3.(int32), val4.(int32), val5.(int32), val6.(float64), loc)
 	},
 }
+
+func getTimestampInServerLocation(year, month, day, hour, minute int32, partialSecond float64, loc *time.Location) (time.Time, error) {
+	if year == 0 {
+		return time.Time{}, errDateFieldOutOfRange
+	} else if year < 0 {
+		// PostgreSQL: year 0 = 1 BC, year -1 = 2 BC, etc.
```

Comment:
```suggestion
		// PostgreSQL: year 0 = 1 BC, year 1 = 2 BC, etc.
```

Reply:
This is actually correct comment, -1 year in time.Time value is formatted to 2 BC